src/vos/tests/evt_ctl.c

```diff
@@ -750,6 +752,347 @@ global_teardown(void **state)
+	int loop_count;
+
+	count = 0;
+	switch (iter_flag) {
+	case EVT_ITER_EMBEDDED:
+		print_message("EVT_ITER_EMBEDDED\n");
+	break;
+	case EVT_ITER_VISIBLE:
+		print_message("EVT_ITER_VISIBLE\n");
+		copy_exp_val_to_array(iter_flag, evt_data, expval, &count);
+	break;
+	case EVT_ITER_COVERED:
+		print_message("EVT_ITER_COVERED\n");
+		count = 1;
+		copy_exp_val_to_array(iter_flag, evt_data, expval, &count);
+	break;
+	case (EVT_ITER_VISIBLE | EVT_ITER_COVERED):
+		print_message("EVT_ITER_VISIBLE (COVERED)\n");
+		count = 1;
+		copy_exp_val_to_array(iter_flag, evt_data, expval, &count);
+	break;
+	case (EVT_ITER_SKIP_HOLES|EVT_ITER_VISIBLE):
+		print_message("EVT_ITER_SKIP_HOLES (VISIBLE)\n");
+		copy_exp_val_to_array(iter_flag, evt_data, expval, &count);
+	break;
+	case (EVT_ITER_REVERSE | EVT_ITER_SKIP_HOLES
+			| EVT_ITER_VISIBLE):
+		print_message("EVT_ITER_REVERSE (SKIP_HOLES and VISIBLE)\n");
+	break;
+	case (EVT_ITER_REVERSE | EVT_ITER_COVERED):
+		 print_message("EVT_ITER_REVERSE (COVERED)\n");
+	break;
+	case (EVT_ITER_REVERSE | EVT_ITER_VISIBLE | EVT_ITER_COVERED):
+		print_message("EVT_ITER_REVERSE (VISIBLE and COVERED)\n");
+	break;
+	case (EVT_ITER_REVERSE | EVT_ITER_VISIBLE):
+		print_message("EVT_ITER_REVERSE (VISIBLE)\n");
+	break;
+	default:
+		print_message("Invalid Flag\n");
+	}
+	if ((iter_flag & EVT_ITER_COVERED) == EVT_ITER_COVERED)
+		expected_size = count-1;
+	else
+		expected_size = count;
+	for (loop_count = expected_size,
+		count = 1; loop_count >= 1;
+		loop_count--, count++) {
+		rev_expval[count] = expval[loop_count];
+	}
+}
+
+/* test_evt_iter_flags :
+*
+* Validate the following conditions:
+*   10: EVT_ITER_VISIBLE|EVT_ITER_SKIP_HOLES
+*   26: EVT_ITER_REVERSE|EVT_ITER_VISIBLE|EVT_ITER_SKIP_HOLES
+*    2: EVT_ITER_VISIBLE
+*   18: EVT_ITER_REVERSE|EVT_ITER_VISIBLE
+*    4: EVT_ITER_COVERED
+*   20: EVT_ITER_REVERSE|EVT_ITER_COVERED
+*    6: EVT_ITER_COVERED|EVT_ITER_VISIBLE
+*   22: EVT_ITER_REVERSE|EVT_ITER_COVERED|EVT_ITER_VISIBLE
+*    1: EVT_ITER_EMBEDDED
+*/
+static void
+test_evt_iter_flags(void **state)
+{
+	struct test_arg		*arg = *state;
+	int			*value;
+	daos_handle_t		 toh;
+	daos_handle_t		 ih;
+	struct evt_entry_in	 entry = {0};
+	struct evt_entry	 ent;
+	int			rc;
+	int			epoch;
+	int			offset;
+	int			sum;
+	int			iter_count;
+	int			count;
+	uint32_t	inob;
+	int		**data;
+	int		*exp_val;
+	int		*actual_val;
+	int		*rev_exp_val;
+	int		hole_epoch;
+	int		val[] = {10, 26, 2, 18, 4, 20, 6, 22, 1};
+	int		t_repeats;
+
+	/* Create a evtree */
+	rc = evt_create(EVT_FEAT_DEFAULT, 13, arg->ta_uma, arg->ta_root,
+			DAOS_HDL_INVAL, &toh);
+	assert_int_equal(rc, 0);
+	D_ALLOC_ARRAY(data, (NUM_EPOCHS+1));
+	if (data == NULL)
+		goto end;
+	for (count = 0; count < NUM_EPOCHS+1; count++) {
+		data[count] = (int *)calloc(
+				(NUM_EPOCHS+NUM_EXTENTS+1), sizeof(int));
+		if (data[count] == NULL) {
+			print_message("Cannot allocate Memory\n");
+			goto end;
+		}
+	}
+	exp_val = (int *)malloc((NUM_EPOCHS+1)*
```

Comment:
sorry, there are a few more malloc/calloc calls here....should have pointed them out.   Looks like all can be D_ALLOC_ARRAY.   There are also D_ALLOC_PTR and D_ALLOC macros...the former does sizeof(*ptr) for you and the latter is for more custom allocations.   In general, if you are using D_FREE, you should use one of these macros.

Reply:
All malloc/calloc use D_ALLOC_ARRAY now...